java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/protocol/match/MatchField.java

```diff
@@ -31,15 +31,24 @@
 import org.projectfloodlight.openflow.types.VxlanNI;
 import org.projectfloodlight.openflow.types.VFI;
 
+import java.util.Collections;
+import java.util.Set;
+import java.util.HashSet;
+
 public class MatchField<F extends OFValueType<F>> {
     private final String name;
     public final MatchFields id;
-    private final Prerequisite<?>[] prerequisites;
+    private final Set<Prerequisite<?>> prerequisites;
```

Comment:
Another one: ImmutableSet automatically will re-use the shared empty instance, so no special code needed for that.